conan/internal/runner/ssh.py

```diff
@@ -119,92 +104,98 @@ def ensure_runner_environment(self):
+        has_remote_conan = self.remote_conn.check_file_exists(conan_cmd)
+
+        if self.is_remote_windows:
+            python_command = remote_folder + "/venv" + "/Scripts" + "/python.exe"
+        else:
+            python_command = remote_folder + "/venv" + "/bin" + "/python"
 
         if not has_remote_conan:
-            _, _stdout, _stderr = self.client.exec_command(f"{python_command} -m venv {conan_venv}")
-            if _stdout.channel.recv_exit_status() != 0:
-                ssh_info(f"Unable to create remote venv: {_stderr.read().decode().strip()}")
+            self.logger.verbose(f"Creating remote venv")
+            result = self.remote_conn.run_command(f"{self.remote_python_command} -m venv {conan_venv}", "Creating remote venv")
+            if not result.success:
+                self.logger.error(f"Unable to create remote venv: {result.stderr}")
+            self._install_conan_remotely(python_command, requested_conan_version)
+        else:
+            version = self.remote_conn.run_command(f"{conan_cmd} --version", "Checking conan version", verbose=True).stdout
+            remote_conan_version = Version(version[version.rfind(" ")+1:])
+            if requested_conan_version == "dev" and remote_conan_version.bump(1) == str(conan_version).replace("-dev", ""):
+                pass
+            elif remote_conan_version != requested_conan_version:
+                self.logger.verbose(f"Remote Conan version mismatch: {remote_conan_version} != {requested_conan_version}")
+                self._install_conan_remotely(python_command, requested_conan_version)
 
-            if is_windows:
-                python_command = remote_folder + "/venv" + "/Scripts" + "/python.exe"
-            else:
-                python_command = remote_folder + "/venv" + "/bin" + "/python"
+        if not self.remote_conn.run_command(f"{conan_cmd} remote update conancenter --url='https://center2.conan.io'", "Updating conancenter remote").success:
+            self.logger.error(f"Unable to update conancenter remote: {result.stderr}")
 
-            _, _stdout, _stderr = self.client.exec_command(f"{python_command} -m pip install git+https://github.com/conan-io/conan@feature/docker_wrapper")
-            if _stdout.channel.recv_exit_status() != 0:
-                # Note: this may fail on windows
-                ssh_info(f"Unable to install conan in venv: {_stderr.read().decode().strip()}")
+        self._create_remote_conan_wrapper(remote_conan_home, remote_folder, conan_cmd)
 
+    def _install_conan_remotely(self, python_command: str, version: str):
+        self.logger.verbose(f"Installing conan version: {version}")
+        # Note: this may fail on windows
+        result = self.remote_conn.run_command(f"{python_command} -m pip install conan{f'=={version}' if version != 'dev' else ' --upgrade'}", "Installing conan")
+        if not result.success:
+            self.logger.error(f"Unable to install conan in venv: {result.stderr}")
+
```

Comment:
I don't think this ssh runner should have the responsibility to install Conan in the remote machine, lets keep it as minimal as possible.

Reply:
This behavior is present in develop2, not a new introduction.
The main idea is that the SSH runner will create a completely isolated environment and a completely isolated Conan installation to avoid any possible interference with the host machine.
Also, the installed conan version should match the caller version.

Reply:
@memsharded this is something that was already in the first version and it removes some friction for 2 reasons.

1. You don't need to connect machine by machine installing each conan version.
2. If you change the conan version in your local machine and use a new feature, it guarantees that the remotes will be aligned with you.

With the docker runners we don't do something similar because you have the option to define a dockerfile and it is something more local, but in this context is more necessary.